modules/rte_mis_state/rte_mis_state.module

```diff
@@ -88,61 +81,52 @@ function rte_mis_state_form_alter(&$form, FormStateInterface $form_state, $form_
     if (isset($form['field_admin_details'])) {
       $form['field_admin_details'] += [
         '#states' => [
-          'visible' => [
+          'invisible' => [
             ':input[name="roles"]' => [
-              ['value' => 'state_admin'],
-              ['value' => 'district_admin'],
-              ['value' => 'block_admin'],
+              ['value' => 'school_admin'],
+              ['value' => 'authenticated'],
             ],
           ],
         ],
       ];
 
-      $form['location_container']['#prefix'] = '<div id="wrapper">';
-      $form['location_container']['#suffix'] = '</div>';
-      $form['location_container']['location_parent'] = [
-        '#type' => CshsElement::ID,
-        '#label' => t('Location'),
-        '#labels' => [
-          'District',
-          'Block',
-        ],
-        '#states' => [
-          'visible' => [
-            ':input[name="roles"]' => [
-              ['value' => 'district_admin'],
-              ['value' => 'block_admin'],
-            ],
+      $form['field_location_details']['#prefix'] = '<div id="wrapper">';
+      $form['field_location_details']['#suffix'] = '</div>';
+
+      $form['field_location_details']['#states'] = [
+        'visible' => [
+          ':input[name="roles"]' => [
+            ['value' => 'district_admin'],
+            ['value' => 'block_admin'],
           ],
         ],
       ];
       // Mapping for different roles.
       $role_mappings = [
         'state_admin' => [
-          "authenticated" => "Authenticated user",
           'district_admin' => 'District Admin',
           'block_admin' => 'Block Admin',
-          'nodal_officer' => 'Nodal Officer',
         ],
         'district_admin' => [
-          "authenticated" => "Authenticated user",
           'block_admin' => 'Block Admin',
-          'nodal_officer' => 'Nodal Officer',
         ],
       ];
 
       // Based on current user role.
       // Check for the user roles they can create account.
-      if (isset($current_user_roles[1])) {
-        if (isset($role_mappings[$current_user_roles[1]])) {
-          // Check the current path.
-          $current_path = \Drupal::service('path.current')->getPath();
-          // Create a custom link.
-          $user_edit_path = Url::fromRoute('entity.user.edit_form', ['user' => $current_user_id])->toString();
-          // Check if its current user's edit page.
-          if ($current_path == $user_edit_path) {
-            // $associated_role me add kar do current user ka role.
-            $form['location_container']['location_parent']['#access'] = FALSE;
+      $current_user_role = $current_user_roles[1];
```

Comment:
Can we null coalescing operator here to avoid an undefined index warning?

Reply:
Added!